app/routes/api/v1/warehouse/products/count/by-type/$productTypeId.ts

```diff
@@ -0,0 +1,28 @@
+import { LoaderFunction } from "@remix-run/node";
+import { countProductByType } from "~/models/product/product.server";
+
+type LoaderData = number;
+
+export const loader: LoaderFunction = async ({
+  params,
+}): Promise<LoaderData> => {
+  const productTypeId = params.productTypeId;
+
+  if (productTypeId == null) {
+    throw new Response(null, {
+      status: 400,
+      statusText: "Product Type Not Provided",
+    });
+  }
+
+  const countedPoducts = await countProductByType(productTypeId);
+
+  if (countedPoducts == null) {
+    throw new Response(null, {
+      status: 404,
+      statusText: "Product Type Not Found",
+    });
+  }
+
+  return countedPoducts;
```

Comment:
`countedPoducts` should be `countedProducts`.

Also, please do not return directly json values, but return a json object wrapping the value, something like:

```json
{
  numberOfProducts: countedProducts
}
```

In order to correctly return a json response, you could use the [`json`](https://remix.run/docs/en/v1/api/remix#json) utility.